hasher-matcher-actioner/hmalib/lambdas/api/datasets_api.py

```diff
@@ -0,0 +1,171 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import bottle
+import typing as t
+from dataclasses import dataclass, asdict
+from hmalib.aws_secrets import AWSSecrets
+from threatexchange.api import ThreatExchangeAPI
+from hmalib.common.logging import get_logger
+from .middleware import jsoninator, JSONifiable
+from hmalib.common.config import HMAConfig
+from hmalib.common import config as hmaconfig
+from hmalib.lambdas.fetcher import ThreatExchangeConfig
+
+logger = get_logger(__name__)
+
+
+@dataclass
+class Dataset(JSONifiable):
+    privacy_group_id: int
+    privacy_group_name: str
+    fetcher_active: bool
+    write_back: bool
+    in_use: bool
+
+    def to_json(self) -> t.Dict:
+        return asdict(self)
+
+    @classmethod
+    def from_dict(cls, d: dict) -> "Dataset":
+        return cls(
+            d["privacy_group_id"],
+            d["privacy_group_name"],
+            d["fetcher_active"],
+            d["write_back"],
+            d["in_use"],
+        )
+
+
+@dataclass
+class DatasetsResponse(JSONifiable):
+    datasets_response: t.List[Dataset]
+
+    def to_json(self) -> t.Dict:
+        return {
+            "datasets_response": [
+                dataset.to_json() for dataset in self.datasets_response
+            ]
+        }
+
+
+@dataclass
+class SyncDatasetResponse(JSONifiable):
+    response: str
+
+    def to_json(self) -> t.Dict:
+        return asdict(self)
+
+
+def sync_privacy_groups():
```

Comment:
nit: This seems to business logic specific to be here. If it is possible for this code to be invoked outside of a lambda, consider moving to its own module outside of the `hmalib.lambdas` space.